EncoreTickets.SDK.CustomRules/EncoreTickets.SDK.CustomRules.Test/CustomRulesTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Threading.Tasks;
+using EncoreTickets.SDK.CustomRules.Enums;
+using EncoreTickets.SDK.CustomRules.Rules;
+using NUnit.Framework;
+using Verify = Microsoft.CodeAnalysis.CSharp.Testing.NUnit.AnalyzerVerifier<
+    EncoreTickets.SDK.CustomRules.CustomRulesAnalyzer
+>;
+
+namespace EncoreTickets.SDK.CustomRules.Test
+{
+    [TestFixture]
+    public class CustomRulesTests
+    {
+        [Test]
+        public async Task ElementOrderingRule_IfRuleIsNotBroken_ReportsNoDiagnostics()
+        {
+            var testCode = @"
+    namespace ConsoleApplication
+    {
+        class TypeName
+        {   
+            public enum Enum {}
+
+            public interface IInterface {}
+
+            public struct Struct {}
+
+            public class Class {}
+
+            public string field;
+
+            public string Property { get; set; }
+
+            public string this[int index] => ""Indexer"";
+
+            public delegate string Delegate();
+
+            public event Delegate Event;
+
+            public TypeName() {}
+
+            public void Method() {}
+
+            ~TypeName() {}
```

Comment:
In the ordering scheme that we discussed, destructors must be after constructors and before methods. And an indexer should also be elsewhere

Reply:
I can put destructor after constructor but I think it makes sense to have indexer after properties instead of having it between destructor and methods since it's basically just a special case of property

Reply:
agree, let's do that